Main menu

FizzySD opened the pull request:
Closes #(ADD ISSUE NUMBER)

**Notes for the reviewer:**

*(Add information which you think is important for the reviewer to be aware of (i.e. functionality X is not included as this was too complex, see issue 123))*

**Contributor guidelines:**
1. A build has been uploaded to the discord #builds channel & has been tested by the testing team
2. Assure the code is inline with the [AoTTG2 coding conventions](https://github.com/AoTTG-2/AoTTG-2/wiki/Code-&-Style-Conventions)
3. Fix any issues that the CI reports (Sonarcloud & unit tests)

After all above conditions are met, @Lithrun will review your Pull Request. Resolve any comments of the reviewer or discuss them if you disagree.

FizzySD committed:
d

FizzySD committed:
Fully imported menu from previus branch

FizzySD committed:
Fixxed animation

FizzySD committed:
Renamed Folders

FizzySD committed:
Imported sound (to be changed)

FizzySD committed:
Small changes

FizzySD committed:
Small fixxes

FizzySD committed:
Animation fixxes and sound additions

FizzySD committed:
Changed ServerSelection UI

FizzySD committed:
Fixxes

FizzySD committed:
Added animaton when server changes

FizzySD committed:
Changes USA image

FizzySD committed:
Added Images to UI

chakung committed:
MainMenu

Fixed bugs animation restart on in-game menu by added an action on quick button that change animator menu-tab from MainMenu > Right Panel

FizzySD committed:
Merge branch 'MainMenu' of https://github.com/AoTTG-2/AoTTG-2 into MainMenu

FizzySD committed:
Updated menu

FizzySD committed:
Updated menu

FizzySD committed:
Finished Main-Menu

FizzySD committed:
Changed a bit the scene

Lithrun commented on AoTTG-22.sln:
```diff
@@ -0,0 +1,74 @@
+﻿
```
Delete this duplicate AoTTG-22.sln

Lithrun commented on AoTTG2.Core.csproj:
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <LangVersion>8.0</LangVersion>
```
C# 8 MUST be used, we can't downgrade to C# 7

Lithrun commented on Assets/MenuBackground/main_menue.prefab:
```diff
@@ -0,0 +1,12852 @@
+%YAML 1.1
```
Typo in prefab, "main_menue" doesn't exist. Should be Main Menu

Lithrun commented on Assets/ModelAndAnim/FizzySD/DK Compagnon.otf.meta:
```diff
@@ -0,0 +1,21 @@
+fileFormatVersion: 2
```
Based on the license of this font: https://www.1001freefonts.com/view-pdf/20019/SGFub2RlZCBGb250cyBMaWNlbnNlICYgRkFRIC0gRE8gUkVBRCBUSElTIS5wZGY=/hanoded-fonts-license---faq---do-read-this-.pdf , we can't use it. So delete it

Lithrun commented on Assets/Photon Unity Networking/Resources/PhotonServerSettings.asset:
```diff
@@ -15,14 +15,14 @@ MonoBehaviour:
   AppID: 
   VoiceAppID: 
   ChatAppID: 
-  HostType: 2
```
Changing the HostType & JoinLobby setting breaks multiplayer, undo these changes

Lithrun reviewed (changes requested):
- Move new files within the "Assets" folder to a dedicated folder
- Move the MenuBackGround folder to the Assets/UI/Main Menu folder
- There seems to be duplicates of various files. Please check if these duplicates are required or not, if they are, give them a proper name instead of "Post Processing 1", if they aren't, delete them
- Rename "USA" to "NA" (North America), as no servers are tied to a country, only to a region. We have: Europe, South America, North America & Asia as our planned servers.
- Remove the "Japan" region, as we don't support this.
- For the regions, you should use the PhotonServerConfig ScriptableObject and extend this with the new customization (I.e.: Image)
- Localization seems to be missing
- Don't use underscores within file names
- "Assets/MenuBackground" & "Assets/ModelAndAnim/FizzySD" have a lot of duplicate files
- Instead of requiring multiple animations for buttons, LeanTween might also be able to do this for you https://assetstore.unity.com/packages/tools/animation/leantween-3595 (its already installed)

FizzySD committed:
Removed Duplicated files

FizzySD committed:
Removed unused animation

FizzySD committed:
Finally deleted all the duplicates files

FizzySD committed:
Extender PhotonServerSettings

FizzySD committed:
Changed SA icon

FizzySD committed:
Added the main menu prefab

since the material got corrupted for some reason, i retextured all

Mi-Sad committed:
Cleaned Unseen unused go in the scene

Removed cameras not rendering, (there were 4 in the scene 2 are enough) and few object out of the view

Mi-Sad committed:
Adding ButtonAnimation to animate menu button

Uses leanTween to animate and play sound on hover.

Mi-Sad committed:
Animated Server Selection and setted scalable main menu to work with different screen ratio

include the addition to ServerSelector class in order to animate the panel through callback. Introduce the mapbuttonanimation class to animate the various region selection as a child of the buttonanimation class, used just to animate the buttons in the menu, also remove some duplicates and some obsolete files and animation of the manu no more used, also clean even more the scene aottg2.

FizzySD committed:
Animation Looped, fix some settings, added sfx

FizzySD committed:
fixxed animation

fixxed animation since i'm dumb and i forgot about saving the scene

Mi-Sad committed:
Added overlay to the buttons and text color animation

PS LeanTween is prone to misbehave so might switch everything to self made animation coroutines

FizzySD committed:
Fixxed some bugs

Mi-Sad committed:
Added script to avoid restarting the animation after quitting a room

because the animator is in the loaded scene and the menu component is on don't destry on load i have to search the object to bypass the animator and set it to idle right away

Mi-Sad committed:
Resizing Render Texture for Main Menu

the Render Texture now depends on the size of the actual screen that has to be rendered, scaling and not stretching the texture

FizzySD committed:
commented malfunctioning code for a  hacky unity bug

Mi-Sad committed:
Merge branch 'development' into MainMenu

Mi-Sad committed:
Merge branch 'development' into MainMenu

Mi-Sad committed:
Fixed Scene For UltraWide Screen

Added function to scale depth of field with the screen size and added scale for screen ratio and field of view to have a more consistent view through the various possible screen size and aspect ratios. Added some details and extended the scene for ultra wide screen (32:9)
Also initalized localization for ServerSelector screen.

Mi-Sad committed:
Removed audio after first time as well

Removed audio played when the menu gets on

Mi-Sad committed:
Removed almost all code smells

Only ones left are the ones caused by serializefield, not recognized by sonarcloud which than thinks the code won't ever be assigned

Mi-Sad committed:
Added a RenderTexture support check

Mi-Sad committed:
RenderTexture creation with getTemporary

Trying to fix the Test in playmode error on sonarcloud by avoiding calling the RenderTexture.Create

Mi-Sad committed:
Added a default renderTexture target

Added a default render texture target in case the system is unable to generate a Render texture target itself

Mi-Sad committed:
Run out of ideas so starting to test random alternative to the code sonarcloud doesn't like

Mi-Sad committed:
Added check to avoid creating render texture if there's no supported

Mi-Sad committed:
Moved all the graphic and quality setting to a dedicated class QualityAdaptator

Added some check on the framerate to in case reduce the size of the renderTexture.

Lithrun committed:
Change RenderTexture format and depth for CI/CD when not supported

Lithrun committed:
Attempt to set Graphics Format

Lithrun committed:
Release the RenderTexture on pipeline

Lithrun committed:
Disable render texture for PlayMode tests

Lithrun committed:
Exclude Adaptor logic

Lithrun committed:
Resolve null reference

Lithrun committed:
Disable mesh renderer by default

Lithrun committed:
Include OnDisable

Lithrun committed:
Update build.yml

Lithrun committed:
Update build.yml

Try without force open gl

Lithrun committed:
Try removing custom params

Lithrun committed:
Revert "Change RenderTexture format and depth for CI/CD when not supported"

This reverts commit 977a4ceb1e262ca1e1602f1150c05e1bfeea4e71.

Lithrun committed:
Merge branch 'MainMenu' of https://github.com/AoTTG-2/AoTTG-2 into MainMenu

Lithrun committed:
Undo commits

Lithrun committed:
Undo commits

Lithrun committed:
Solve cursor issue
